core/handler/downlink.go

```diff
@@ -69,22 +69,17 @@ func (h *handler) HandleDownlink(appDownlink *mqtt.DownlinkMessage, downlink *pb
 	// Run Processors
 	for _, processor := range processors {
 		err = processor(ctx, appDownlink, downlink)
-		// If the payload hasn't to be sent
+		// If the payload hasn't to be sent (bad payload)
```

Comment:
It's not necessarily bad, but there might be a reason not to emit it. We don't know the reason here. The `ErrNotNeeded` check is clear